subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dependencies/DefaultMutableVersionConstraint.java

```diff
@@ -87,6 +88,14 @@ public void strictly(String version) {
         doStrict();
     }
 
+    @Override
+    public void reject(String... versions) {
+        if (versions == null) {
+            throw new InvalidUserDataException("The 'reject' clause requires at least one rejected version");
```

Comment:
This sounds like you want to check `versions.empty`?

:o: I am not sure if we need this check at all. But yes, it can be helpful as calling reject without an argument has no effect and should not be done. So I would check `versions.empty` (we can check for null in addition, but that should not happen unless someone explicitly call the method with null). 
Then we should also add a test that checks the error.

Reply:
No. It's a DSL method. If you call `reject`, since it's a varargs method, `versions` will be `null`.

Reply:
Alright. My knowledge about that is a little thin it seems. Can we still add small integration test that triggers the error?

Reply:
Actually you were right, `null` is not possible here in Java. I added a unit test for this, I don't think it's worth paying the price of an integration test here.